omod/src/main/java/org/openmrs/module/smartonfhir/web/servlet/SmartLaunchOptionSelected.java

```diff
@@ -76,15 +88,34 @@ private String getParameter(HttpServletRequest request, String parameter) {
 		return result;
 	}
 	
-	private String getParameterFromStringUrl(String url, String parameter) {
-		List<NameValuePair> decodedUrlParams = URLEncodedUtils.parse(url, StandardCharsets.UTF_8);
+	private String getParameterFromStringUrl(String url, String parameter) throws URISyntaxException {
+		MultiValueMap<String, String> params = UriComponentsBuilder.fromUriString(url).build().getQueryParams();
 		
-		for (NameValuePair obj : decodedUrlParams) {
-			if (obj.getName().equals(parameter)) {
-				return obj.getValue();
+		if (params.containsKey(parameter)) {
+			return params.getFirst(parameter);
+		}
+		
+		return null;
+	}
+	
+	private String getLaunchTypeString(String key) {
+		JsonWebToken appToken;
+		
+		try {
+			appToken = TokenVerifier.create(key, JsonWebToken.class).getToken();
+		}
+		catch (VerificationException e) {
+			e.printStackTrace();
```

Comment:
This should probably be:

```java
log.error("Verification exception while trying to determine launchType", e);
```

Reply:
It might actually be better to just not catch the verification exception here and instead handle it in `doGet()`.